lib/matplotlib/axes/_axes.py

```diff
@@ -5843,6 +5846,8 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
                     xvals.append(x.copy())
                     yvals.append(y.copy())
 
+            fill_kwargs = dict() if not input_empty else dict(linewidth=0)
```

Comment:
Again. I think showing the line is the correct thing to do.  My main motivation is that if some one now wants to update the histogram artists, this will cause a great deal of trouble.